USER_GUIDE.md

```diff
@@ -64,6 +66,34 @@ Transport transport = new RestClientTransport(restClient,
 OpenSearchClient client = new OpenSearchClient(transport);
 ```
 
+### Ignore certificate check when running OpenSearch as docker container with self-signed certificate
+
+When running OpenSearch via [docker](https://opensearch.org/docs/latest/opensearch/install/docker/), container uses
+a self-signed certificate can trigger `javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target`.
+If you don't want to import the generated cert, you can ignore the certificate validation in the client.
+
+```java
+public RestClient createLocalRestClient() {
```

Comment:
@pingleig this is not the right repository for the change: the `RestClient` is part of the core [1], please check the documentation snippets in the `RestClientDocumentation` [2] and make your contribution there please. 

[1] https://github.com/opensearch-project/OpenSearch
[2] https://github.com/opensearch-project/OpenSearch/blob/main/client/rest/src/test/java/org/opensearch/client/documentation/RestClientDocumentation.java

Please be aware that: 
 - the `main` branch is next `3.0.0` and uses Apache HttpClient 5.x
 - the `2.x`  branch uses   Apache HttpClient 4.x

The snippets will be different for each of those. Thank you.